benchto-driver/src/main/java/com/teradata/benchto/driver/graphite/GraphiteMetricsLoader.java

```diff
@@ -72,10 +73,10 @@ public void initQueryMetrics()
     }
 
     @Override
-    public List<Measurement> loadMeasurements(Measurable measurable)
+    public CompletableFuture<List<Measurement>> loadMeasurements(Measurable measurable)
```

Comment:
Letting measurement  provider return data after delay is the cornerstone of this change. If I don't "future-ize" this, I would need to couple the delay-related-logic in the calling code.